src/freenet/clients/http/FProxyToadlet.java

```diff
@@ -949,7 +949,7 @@ public void run() {
 					}
 					option = optionList.addChild("li");
 					// FIXME: is this safe? See bug #131
-					MediaType textMediaType = new MediaType("text/plain");
+					MediaType textMediaType = new MediaType("text/plain;charset=utf-8");
 					textMediaType.setParameter("charset", (e.getExpectedMimeType() != null) ? MediaType.getCharsetRobust(e.getExpectedMimeType()) : null);
```

Comment:
It kinda looks like this line would immediately clobber whatever you’ve set in the constructor…

Reply:
No, this would have been easier if you had written a test first. 😄 

(I’m currently trying to write one, but this would be easier if somebody merged https://github.com/hyphanet/fred/pull/1043 first, because that one contains a lot of shit for easier tests. 🙂)

Reply:
Do you really mean the "quote n2ntm PR"?

(it feels like the quoting PR still needs so much work that I should only finish it after the next release)